### PR TITLE
Upgrade to latest TC version

### PR DESCRIPTION
**Pull request by charleskawczynski**
Whoohoo! This speeds things up a bit 🚀

Wow, there's so much variation between runs!

### CODE REVIEW

File: src/TurbulenceConvectionUtils.jl [hunk 1]
```diff
@@ -234,12 +234,12 @@ function run_reference_SCM(
         if m.case_name == "LES_driven_SCM"
             namelist["meta"]["lesfile"] = get_stats_path(y_dir(m))
         end
+        namelist["stats_io"]["calibrate_io"] = true
```
Comment (ilopezgp) [inline src/TurbulenceConvectionUtils.jl:237]: I prefer if we set this before the optional args,

```
# Set optional namelist args
        if !isnothing(namelist_args)
            for namelist_arg in namelist_args
                change_entry!(namelist, namelist_arg)
            end
        end
```

This way, we can change `calibrate_io` to false if we want to. Same in the other instance of this.

File: src/TurbulenceConvectionUtils.jl [hunk 2]
```diff
@@ -307,18 +307,17 @@ function run_SCM_handler(
...
-    catch e
+    _, ret_code = Logging.with_logger(Logging.NullLogger()) do
+        main1d(namelist; time_run = false)
+    end
+    if ret_code ≠ :success
         model_error = true
         message = ["TurbulenceConvection.jl simulation $(basename(m.y_dir)) failed with parameters: \n"]
         append!(message, ["$param_name = $param_value \n" for (param_name, param_value) in zip(u_names, u)])
         @warn join(message)
-        @warn join(["This was caused by ", e])
```
Comment (ilopezgp) [inline src/TurbulenceConvectionUtils.jl:321]: Is there a way we can report the error message from TC.jl? Like this, we only know that the simulation failed and for what parameters, but not the actual error.

Comment (charleskawczynski) [reply src/TurbulenceConvectionUtils.jl:321]: I guess this goes along the lines of what Jake was suggesting with re-throwing the error message? We can use the return code to print something, but I guess the new try-catch in TC may swallow errors.. Maybe we need to update that..

Comment (ilopezgp) [reply src/TurbulenceConvectionUtils.jl:321]: I think that would be really helpful.

Comment (charleskawczynski) [reply src/TurbulenceConvectionUtils.jl:321]: This is technically an upstream issue-- we just need for TC to add this to the warn log. I'll open a PR, but I don't think that should stop this PR.

File: src/TurbulenceConvectionUtils.jl (hunk 1 above)
Comment (charleskawczynski) [reply src/TurbulenceConvectionUtils.jl:237]: I see two places where this exists, do you want me to change both to 
```julia
    namelist["stats_io"]["calibrate_io"] = true
    # Set optional namelist args
    if !isnothing(namelist_args)
        for namelist_arg in namelist_args
            change_entry!(namelist, namelist_arg)
        end
    end
```
If not, do you want to push a commit of what you had in mind?

Comment (ilopezgp) [reply src/TurbulenceConvectionUtils.jl:237]: Yes! That works.

Comment (ilopezgp) [reply src/TurbulenceConvectionUtils.jl:238]: I think now the line is added twice

Comment (charleskawczynski) [reply src/TurbulenceConvectionUtils.jl]: Whoops! Fixed.

**Review by ilopezgp — APPROVED** [review]

File: src/TurbulenceConvectionUtils.jl (hunk 1 above)
Comment (charleskawczynski) [reply src/TurbulenceConvectionUtils.jl]: @ilopezgp, we still need `namelist["stats_io"]["calibrate_io"] = true` in the tests, it looks like, correct?

Comment (charleskawczynski) [reply src/TurbulenceConvectionUtils.jl]: I added that to the one unit test.

**Pull request merged**